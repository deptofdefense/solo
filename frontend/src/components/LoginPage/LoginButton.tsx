import React from "react";
import classes from "./LoginPage.module.scss";

interface LoginButtonProps {
  onClick: React.MouseEventHandler;
}

const LoginButton: React.FC<LoginButtonProps> = ({ onClick }) => (
  <div className={classes.btnRoot}>
    <button className="usa-button usa-button--big" onClick={onClick}>
      CAC Login
    </button>
  </div>
);

export default LoginButton;
