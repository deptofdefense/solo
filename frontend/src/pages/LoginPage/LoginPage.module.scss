.root {
  // margin-top: 10%;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.btnRoot {
  margin-top: 25px;
}

.errorMsg {
  color: #cc0000;
  display: block;
  margin-top: 1.5em;
}

.consent {
  max-width: 600px;
  padding: 30px 40px;
  background-color: #ffffcc;

  ul {
    line-height: 1.3rem;
  }
}
