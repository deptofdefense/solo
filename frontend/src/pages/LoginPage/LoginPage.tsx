import React, { useState } from "react";
import useAuthContext from "context/AuthContext";
import { Button } from "solo-uswds";
import classes from "./LoginPage.module.scss";

const LoginPage: React.FC = props => {
  const [loginError, setLoginError] = useState("");
  const { apiLogin } = useAuthContext();

  const login = async () => {
    try {
      await apiLogin();
    } catch (e) {
      setLoginError(
        "Hey Marine. Having trouble logging in? Try closing your browser window."
      );
    }
  };

  return (
    <div className={classes.root}>
      <h1 className={classes.title}>
        Welcome to the System for Operational Logistics Ordering (SOLO)
      </h1>
      <div className={classes.consent}>
        <h3>
          You are accessing a U.S. Government (USG) Information System (IS) that
          is provided for USG-authorized use only.
        </h3>
        <p>
          By using this IS (which includes any device attached to this IS), you
          consent to the following conditions:
        </p>
        <ul>
          <li>
            The USG routinely intercepts and monitors communications on this IS
            for purposes including, but not limited to, penetration testing,
            COMSEC monitoring, network operations and defense, personnel
            misconduct (PM), law enforcement (LE), and counterintelligence (CI)
            investigations.
          </li>
          <li>
            At any time, the USG may inspect and seize data stored on this IS.
          </li>
          <li>
            Communications using, or data stored on, this IS are not private,
            are subject to routine monitoring, interception, and search, and may
            be disclosed or used for any USG-authorized purpose.
          </li>
          <li>
            This IS includes security measures (e.g., authentication and access
            controls) to protect USG interests -- not for your personal benefit
            or privacy.
          </li>
          <li>
            Notwithstanding the above, using this IS does not constitute consent
            to PM, LE or CI investigative searching or monitoring of the content
            of privileged communications, or work product, related to personal
            representation or services by attorneys, psychotherapists, or
            clergy, and their assistants. Such communications and work product
            are private and confidential. See User Agreement for details.
          </li>
        </ul>
      </div>
      <div>
        <Button big className={classes.btnRoot} onClick={login}>
          Agree and CAC Login
        </Button>
      </div>
      {loginError && (
        <div>
          <span className={classes.errorMsg}>{loginError}</span>
        </div>
      )}
    </div>
  );
};

export default LoginPage;
