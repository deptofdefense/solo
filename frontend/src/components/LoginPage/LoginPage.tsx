import React, { useState } from "react";
import useAuthContext from "context/AuthContext";
import LoginButton from "./LoginButton";
import classes from "./LoginPage.module.scss";

const LoginPage: React.FC = props => {
  const [loginError, setLoginError] = useState("");
  const { apiLogin } = useAuthContext();

  const login = async () => {
    try {
      await apiLogin();
    } catch (e) {
      setLoginError(
        "Hey Marine. Having trouble logging in? Try closing your browser window."
      );
    }
  };

  return (
    <div className={classes.root}>
      <h1 className={classes.title}>
        Welcome to the System for Operational Logistics Ordering (SOLO)
      </h1>
      <div>
        <LoginButton onClick={login} />
      </div>
      {loginError && (
        <div>
          <span className="text-red">{loginError}</span>
        </div>
      )}
    </div>
  );
};

export default LoginPage;
