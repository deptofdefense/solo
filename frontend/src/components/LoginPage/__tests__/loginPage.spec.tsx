import React from "react";
import { render, fireEvent, wait } from "test-utils";
import LoginPage from "../LoginPage";

describe("Home page component", () => {
  const fakeLogin = jest.fn();
  const renderOpts = {
    authContext: {
      apiLogin: fakeLogin
    }
  };

  afterEach(() => {
    fakeLogin.mockReset();
  });

  it("matches snapshot", async () => {
    const { asFragment } = render(<LoginPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("calls login on button click", async () => {
    const { getByText } = render(<LoginPage />, renderOpts);
    const loginBtn = getByText("CAC Login");
    fireEvent.click(loginBtn);
    expect(fakeLogin).toHaveBeenCalled();
  });

  it("displays login information on failed login", async () => {
    fakeLogin.mockRejectedValue(new Error("test error message"));
    const { getByText, queryByText } = render(<LoginPage />, renderOpts);
    const loginBtn = getByText("CAC Login");
    fireEvent.click(loginBtn);
    await wait(() => {
      expect(fakeLogin).toHaveBeenCalled();
      expect(queryByText(/Having trouble/)).toBeInTheDocument();
    });
  });
});
