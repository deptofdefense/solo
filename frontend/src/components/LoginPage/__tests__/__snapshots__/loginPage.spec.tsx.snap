// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <h1
      class="title"
    >
      Welcome to the System for Operational Logistics Ordering (SOLO)
    </h1>
    <div>
      <div
        class="btnRoot"
      >
        <button
          class="usa-button usa-button--big"
        >
          CAC Login
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
